From 3f099e18294a5fa5051fe5ae28a58af3c76aee53 Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Mon, 9 Mar 2009 10:50:39 +0800
Subject: [PATCH] ENGR00108890 [MX37] Some micro sd cards can't be recognized.

Reconfigure the SD/MMC pins' PAD to level up the hw timing
compatible.

Signed-off-by: Richard Zhu <r65037@freescale.com>
(cherry picked from commit 3fbaee3ad0c3f95f4f7da87b12a738e51f61606e)
---
 arch/arm/mach-mx37/mx37_3stack_gpio.c |   16 +++++++++-------
 1 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-mx37/mx37_3stack_gpio.c b/arch/arm/mach-mx37/mx37_3stack_gpio.c
index 175aa54..5dd9393 100644
--- a/arch/arm/mach-mx37/mx37_3stack_gpio.c
+++ b/arch/arm/mach-mx37/mx37_3stack_gpio.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2007-2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2007-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -342,6 +342,8 @@ EXPORT_SYMBOL(gpio_activate_audio_ports);
  */
 void gpio_sdhc_active(int module)
 {
+	unsigned int pad_val;
+
 	switch (module) {
 	case 0:
 		mxc_request_iomux(MX37_PIN_SD1_CLK,
@@ -359,37 +361,37 @@ void gpio_sdhc_active(int module)
 
 		mxc_iomux_set_pad(MX37_PIN_SD1_CMD,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_PULL |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_CLK,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_KEEPER |
+				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_DATA0,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_KEEPER |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_DATA1,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_KEEPER |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_DATA2,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_KEEPER |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_DATA3,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_PULL |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
-- 
1.5.4.4

